DocumentsFromSnapshotMigration/src/test/java/com/rfs/FullTest.java

```diff
@@ -177,6 +182,8 @@ public void testDocumentMigration(SearchClusterContainer.Version baseSourceImage
             } finally {
                 deleteTree(tempDir);
             }
+        } finally {
+            executorService.shutdown();
```

Comment:
This is really ugly with how many layers of 'cleanup' is happening, can you abstract or group together the clean up?  As we were walking over the test case today it is really hard to follow